source/guides/git.rst

```diff
@@ -14,153 +14,117 @@ For example, you're trying to develop a feature on Scilpy and your collegue is d
+   :scale: 25 %
    :align: right
 
-.. code-block:: bash
+Let's say you want to develop Scilpy. Instead of cloning Scilpy directly on your computer, you should **fork** it, i.e. copy it as one of your own repositories in your own account. On Github, you can find the button *fork* on the top right of Scilpy's repository page. Then, you can clone this copy of the repository on your computer (find the link by clicking on Github's green button *Code*): :bash:`git clone INSERT_LINK_TO_YOUR_FORKED_REPOSITORY`
 
-    # Cloning the code
-    git clone ${link_to_my_fork}
-    cd my_fork/
+The goal will be to work on your computer and save regularly your changes to **your** fork (it will automatically be named origin). We recommend immediately going inside the cloned folder (:bash:`cd scilpy` in this case) and setting up a remote named *upstream* to help link your fork to the original repository [1]_: :bash:`git remote add upstream git@github.com:scilus/scilpy.git`
 
-    # Telling git who are the upstream and origin
-    git remote add upstream ${link_to_upstream}
-    git remote add origin ${link_to_my_fork} # Should be set automatically
-    git remote -v # To verify everything.
+By default all repositories have a main version of the code (a branch called *master* or *main*). When you are exploring new pieces of code, fixing bugs or tinkering in general you should always create a branch first (with a meaningful name) to avoid messing up your main branch with your tests: :bash:`git checkout -b my_branch`
 
-    # Creating branches
-    git branch my_branch_1 # create it
-    git checkout my_branch_1 # switch on it
-    git checkout -b my_branch_2 # create AND switch
-    # Note that here, my_branch_2 would be a copy of my_branch_1.
-    # If instead you want to start from the master, switch to your master branch first.
+Your main branch in **your** fork should always stay the same as the main branch in the **upstream**. A good practice is to never use :bash:`git add` or :bash:`git commit` in your main branch, only :bash:`git pull upstream master` (or main).
 
-    # Delete a branch:
-    git branch -d my_branch # For a local branch
-    git push origin --delete my_branch # For a branch on origin
+When working in a branch other than the main one (*master* or *main*), you should often save (update) your modifications online. Each update is called a **commit**. First, to see which files have been modified without telling Git, use :bash:`git status`. Then, to add or update a file in Git's memory, use :bash:`git add my_file`. You can also add all files that have been modified and that Git already knew about, with :bash:`git add -u`. If you check again with :bash:`git status`, everything in green will be recorded with the next command: :bash:`git commit -m "Message to explain the work I have been doing in this update"`. To send these changes to the same branch name in your forked repository, use :bash:`git push origin my_branch`.
```

Comment:
Could we explain that a commit is not automatically backed up online, it is only a checkpoint made locally until it is pushed.